apps/mobile-wallet/src/features/addressesManagement/AddressDetailsModal.tsx

```diff
@@ -32,9 +31,8 @@ const AddressDetailsModal = withModal<AddressDetailsModalProps>(({ id, addressHa
   const navigation = useNavigation<NavigationProp<SendNavigationParamList | RootStackParamList>>()
   const dispatch = useAppDispatch()
 
-  const address = useAppSelector((s) => selectAddressByHash(s, addressHash))
   const selectAddressTokens = useMemo(makeSelectAddressesTokens, [])
-  const addressTokens = useAppSelector((s) => selectAddressTokens(s, addressHash))
+  const hasTokens = useAppSelector((s) => selectAddressTokens(s, addressHash)).length > 0
```

Comment:
This way, the component will rerender only when the `number` changes. Before, it would change when anything inside the `addressTokens` array would change.